example/exampleMapD3Crossfilter.html

```diff
@@ -0,0 +1,745 @@
+      // transform the data from the response to be compatible with mapd3.Chart().setData()
+      let data = await transformData(response)
+      return data
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  // currently mapd3 expects data in a particular data structure, so we need to process the "raw" data returned by the database to match it
+  function transformData(_data) {
+    var series = [
+      {
+        group: 0, // will be non-zero for 2nd axis
+        id: 0,
+        label: "count(*)",
+        dimensionName: "Departure Time",
+        measureName: "Number Records",
+        values: _data
+          .map(d => ({
+            key: Array.isArray(d.key0) ? d.key0[0] : d.key0,
+            value: d.val
+          }))
+          .reverse()
+      },
+      {
+        group: 1,
+        id: 1,
+        label: "avg(arrdelay)",
+        dimensionName: "Departure Time",
+        measureName: "Average Arrival Delay",
+        values: _data
+          .map(d => ({
+            key: Array.isArray(d.key0) ? d.key0[0] : d.key0,
+            value: d.val1
+          }))
+          .reverse()
+      }
+    ]
+    return { series }
+  }
+
+  // render combo chart when data is formatted
+  function renderComboChart (dataTransformed) {
+    comboChart.setConfig({
+      parentNode: parentNode,
+      margin: {
+        top: 32,
+        right: 32,
+        bottom: 64,
+        left: 70
+      },
+      width: "auto",
+      height: "auto",
+      palette: null,
+      chartType: "line",
+      xAxisLabel: "Departure Time",
+      yAxisLabel: "Count",
+      y2AxisLabel: "Average Arrival Delay",
+      yDomainEditorIsEnabled: false,
+      y2DomainEditorIsEnabled: false,
+      legendIsEnabled: false,
+      tooltipIsEnabled: true,
+      legendTitle: "",
+      brushIsEnabled: true,
+      brushRangeMin: null,
+      brushRangeMax: null,
+      binExtent: [],
+      rangeBrushExtent: [],
+      timeBin: null,
+      extract: null,
+      autoBin: null,
+      brushExtent: [],
+      keyType: "time",
+      yTicks: "auto",
+      xDomain: "auto",
+      yDomain: "auto",
+      y2Domain: "auto",
+      xLock: false,
+      yLock: false,
+      y2Lock: false,
+      hasRightAxis: false,
+      hasLeftAxis: true,
+      yAxisFormat: "auto",
+      y2AxisFormat: "auto",
+      tooltipFormat: "auto",
+      tooltipTitleFormat: "auto",
+      xAxisFormat: "auto",
+      measureFormats: [],
+      dimensionFormats: []
+    })
+    .setData(dataTransformed)
+  }
+
+  // date object formatter for constructing SQL query from combo chart brushing
+  const dateFormatter = mapd3.d3.timeFormat("%Y-%m-%d %H:%M:%S")
+
+  // set up a few event listeners for the comboChart's brush events
+  comboChart
+    // we can access the combo chart's event listeners using .getEvents()
+    .getEvents()
+    // note: it's a good idea to "throttle" the brushMove event as it gets called very frequently
+    .onBrush("brushMove", throttle(function() {
```

Comment:
Good point. We could do it internally as we already do with mousemove.

Reply:
true. seems like a sensible default behavior to have built into mapd3.